cordapp-cpk/src/test/resources/external-cordapp/build.gradle

```diff
@@ -0,0 +1,61 @@
+plugins {
+    id 'net.corda.plugins.cordapp-cpk'
+}
+
+evaluationDependsOnChildren()
+
+allprojects { p ->
+    println(p.path)
+
+    if (p.path in [rootProject.path, ":corda-platform-cordapp", ":external-cordapp-transitive-dependency"]) {
+        apply plugin: 'java-library'
+        apply plugin: 'maven-publish'
```

Comment:
You need to publish the `corda-api` project too, if you expect
```gradle
dependencies {
   cordaProvided project(':corda-api')
}
```
to work. Although best _not_ to modify `group`, `version` or the `jar` task for the `corda-api` module.

Reply:
I do not understand why, especially considering it is only needed at compile time of this project (I was expecting the `cordaProvided` configuration to not translate to consumer projects but maybe I'm wrong here)

Reply:
for now I simply removed the dependency on `corda-api` as publishing that would have required to change all the test cases that make use of it (due to the missing `maven.repository.dir` project property), tests are green but I don't think the current behavior of the `cordapp-cpk` plugin is correct

Reply:
> for now I simply removed the dependency on `corda-api` as publishing that would have required to change all the test cases that make use of it (due to the missing `maven.repository.dir` project property), tests are green but I don't think the current behavior of the `cordapp-cpk` plugin is correct

You don't have to modify the `corda-api` project _in any way_ in order to publish it. The only reason you weren't publishing it before was because of this filter in your `allprojects {}` block:
```gradle
allprojects { p ->
    println(p.path)

    if (p.path in [rootProject.path, ":corda-platform-cordapp", ":external-cordapp-transitive-dependency"]) {
        // DELIBERATELY ChOOSING NOT TO PUBLISH ':corda-api'
    }
}
```
Whereas something like this in `external-cordapp/build.gradle` would have published `:corda-api` just fine:
```gradle
allprojects {
    apply plugin: 'maven-publish'

    publishing {
        repositories {
            maven {
                name = 'Test'
                url = getProperty("maven.repository.dir")
            }
        }
        publications {
            maven(MavenPublication) {
                from components.java
                pluginManager.withPlugin('net.corda.plugins.cordapp-cpk') {
                    artifact tasks.named('cpk')
                }
            }
        }
    }
}
```